app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt

```diff
@@ -83,11 +114,26 @@ object NetworkModule : InjectionModule {
         }
     }
 
-    private fun Scope.getClient(tag: String, interceptor: Interceptor? = null): OkHttpClient {
+    private fun Scope.getClient(tag: String, resources: Resources, interceptor: Interceptor? = null): OkHttpClient {
         return OkHttpClient.Builder()
             .readTimeout(DEFAULT_CONNECT_TIMEOUT_SECONDS, TimeUnit.SECONDS)
             .connectTimeout(DEFAULT_READ_TIMEOUT_SECONDS, TimeUnit.SECONDS)
             .apply {
+                val appFeatureFlags: AppFeatureFlags = get()
+                if (appFeatureFlags.isSslPinningEnabled) {
+                    try {
+                        createCertificate(resources.openRawResource(R.raw.cert))?.let { sslContext ->
+                            systemDefaultTrustManager()?.let { trustManager ->
```

Comment:
have you tried a newer approach?

https://stackoverflow.com/questions/24006545/how-can-i-pin-a-certificate-with-square-okhttp

I can see your algorithm here is for older (2.x) versions of okhttp

Reply:
I can try it, if you want